cms/search/tests/test_publishers.py

```diff
@@ -0,0 +1,380 @@
+        """Ensure that for a release-type page, release-specific fields get added."""
+        release_calendar_pages = [
+            self.release_calendar_page_provisional,
+            self.release_calendar_page_confirmed,
+        ]
+
+        for page in release_calendar_pages:
+            message = self.publisher._construct_message_for_create_update(page)  # pylint: disable=W0212
+            expected_type = EXPECTED_CONTENT_TYPES[type(page).__name__]
+
+            self.assertEqual(message["content_type"], expected_type)
+            self.assertEqual(message["title"], page.title)
+            self.assertEqual(message["summary"], page.summary)
+            self.assertEqual(message["uri"], page.url_path)
+
+            self.assertIsNotNone(message["release_date"])  # because it's a release
+
+            self.assertEqual(message["release_date"], page.release_date.isoformat())
+
+            self.assertIn("finalised", message)
+            self.assertIn("cancelled", message)
+            self.assertIn("published", message)
+
+            self.assertTrue(message["finalised"])
+            self.assertFalse(message["published"])
+            self.assertFalse(message["cancelled"])
+
+    def test_construct_message_for_release_page_confirmed_date_change(self):
+        """Ensure that for a release-type page, release-specific fields get added."""
+        page = self.release_calendar_page_confirmed
+        page.changes_to_release_date = [
+            {
+                "type": "date_change_log",
+                "value": {"previous_date": timezone.now() - timedelta(days=5), "reason_for_change": "The reason"},
+            }
+        ]
+
+        message = self.publisher._construct_message_for_create_update(page)  # pylint: disable=W0212
+        expected_type = EXPECTED_CONTENT_TYPES[type(page).__name__]
+
+        self.assertEqual(message["content_type"], expected_type)
+        self.assertEqual(message["title"], page.title)
+        self.assertEqual(message["summary"], page.summary)
+        self.assertEqual(message["uri"], page.url_path)
+
+        self.assertIsNotNone(message["release_date"])  # because it's a release
+
+        self.assertEqual(message["release_date"], page.release_date.isoformat())
+
+        self.assertIn("finalised", message)
+        self.assertIn("cancelled", message)
+        self.assertIn("published", message)
+
+        self.assertTrue(message["finalised"])
+        self.assertFalse(message["published"])
+        self.assertFalse(message["cancelled"])
+
+        self.assertIn("date_changes", message)
+
+        self.assertEqual(len(message["date_changes"]), 1)
+
+        date_change = message["date_changes"][0]
+        self.assertEqual(date_change["change_notice"], page.changes_to_release_date[0].value["reason_for_change"])
+        self.assertEqual(
+            date_change["previous_date"],
+            page.changes_to_release_date[0].value["previous_date"].isoformat(),
+        )
```

Comment:
Let's update this to test multiple, not just the first one.

Reply:
Done + added tests for testing topics (topic tagging) also.